macros/scripts/audio.asm

```diff
@@ -1,21 +1,42 @@
-musicheader: MACRO
-	; number of tracks, track idx, address
-	dbw ((\1 - 1) << 6) + (\2 - 1), \3
+channel_count: MACRO
+num_channels = \1 - 1
+ENDM
+
+channel: MACRO
+	dn (num_channels << 2), \1 - 1 ; channel id
+	dw \2 ; address
+num_channels = 0
 ENDM
 
 note: MACRO
-	dn (\1), (\2) - 1
+	dn (\1), (\2) - 1 ; pitch, length
+ENDM
+
+dnote: MACRO
+	note \1, \2 ; drum instrument, length
+ENDM
+
+rest: MACRO
+	note 0, \1 ; length
 ENDM
 
-sound: MACRO
-	note \1, \2
-	db \3 ; intensity
+square_note: MACRO
+	db \1 ; length
+	IF \3 < 0
+		db (\2 << 4) | (%1000 | (\3 * -1)) ; volume envelope
```

Comment:
Are you sure this won't overflow into the high four bits of the macro?
Is this actually used anywhere? Can't you remove the whole `IF` with just
`& 0xF` on the argument?
(same applies to `noise_note` and all other macros that use this construct)

Reply:
By overflow, do you mean between nibbles? It won't overflow if `\3` is in the correct range [-7, 8]

A simple `& 0xF` will not work because `\3` is not in two's compliment, it is in signed magnitude representation.

Reply:
But I admit, it should use `dn`. I just copied it from pokered

Reply:
Fixed these to use `dn` instead of `db`